src/mahalanobis.jl

```diff
@@ -1,11 +1,65 @@
 # Mahalanobis distances
 
+"""
+    Mahalanobis(Q, skipchecks = false) <: Metric
+
+Create a Mahalanobis distance (i.e., a bilinear form) with covariance matrix `Q`.
+Upon construction, both symmetry/self-adjointness and positive semidefiniteness are checked,
+unless the keyword argument `skipchecks = true`.
+
+# Example:
+```julia
+julia> A = collect(reshape(1:9, 3, 3)); Q = A'A;
+
+julia> dist = Mahalanobis(Q)
+Mahalanobis{Matrix{Int64}}([14 32 50; 32 77 122; 50 122 194])
+
+julia> dist = Mahalanobis(A)
+ERROR: ArgumentError: bilinear form is not symmetric/Hermitian
+
+julia> dist = Mahalanobis(A, skipchecks = true)
+Mahalanobis{Matrix{Int64}}([1 4 7; 2 5 8; 3 6 9])
+"""
 struct Mahalanobis{M<:AbstractMatrix} <: Metric
     qmat::M
+    function Mahalanobis(Q::AbstractMatrix; skipchecks::Bool = false)
+        if !skipchecks
+            ishermitian(Q) || throw(ArgumentError("bilinear form is not symmetric/Hermitian"))
```

Comment:
Why not just say "matrix" instead of the more obscure "bilinear form"?

Reply:
Yeah, why not.